contracts/scripts/foundry/InitializeL1ScrollOwner.s.sol

```diff
@@ -0,0 +1,248 @@
+    address L1_ENFORCED_TX_GATEWAY_PROXY_ADDR = vm.envAddress("L1_ENFORCED_TX_GATEWAY_PROXY_ADDR");
+
+    ScrollOwner owner;
+
+    function run() external {
+        vm.startBroadcast(L1_DEPLOYER_PRIVATE_KEY);
+
+        owner = ScrollOwner(payable(L1_SCROLL_OWNER_ADDR));
+
+        // @note we don't config 14D access, since the default admin is a 14D timelock which can access all methods.
+        configProxyAdmin();
+        configScrollChain();
+        configL1MessageQueue();
+        configL1ScrollMessenger();
+        configEnforcedTxGateway();
+        configL2GasPriceOracle();
+        configMultipleVersionRollupVerifier();
+        configL1GatewayRouter();
+        configL1CustomERC20Gateway();
+        configL1ERC721Gateway();
+        configL1ERC1155Gateway();
+
+        // @note comments out for testnet
+        // configL1USDCGateway();
+
+        grantRoles();
+        transferOwnership();
+
+        vm.stopBroadcast();
+    }
+
+    function transferOwnership() internal {
+        Ownable(L1_PROXY_ADMIN_ADDR).transferOwnership(address(owner));
+        Ownable(L1_SCROLL_CHAIN_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_MESSAGE_QUEUE_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_SCROLL_MESSENGER_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_ENFORCED_TX_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L2_GAS_PRICE_ORACLE_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_MULTIPLE_VERSION_ROLLUP_VERIFIER_ADDR).transferOwnership(address(owner));
+        Ownable(L1_GATEWAY_ROUTER_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_CUSTOM_ERC20_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_ETH_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_STANDARD_ERC20_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        // Ownable(L1_USDC_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_WETH_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_ERC721_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+        Ownable(L1_ERC1155_GATEWAY_PROXY_ADDR).transferOwnership(address(owner));
+    }
+
+    function grantRoles() internal {
+        owner.grantRole(SECURITY_COUNCIL_NO_DELAY_ROLE, SECURITY_COUNCIL_ADDR);
+        owner.grantRole(SCROLL_MULTISIG_NO_DELAY_ROLE, SCROLL_MULTISIG_ADDR);
+        owner.grantRole(TIMELOCK_1DAY_DELAY_ROLE, L1_1D_TIMELOCK_ADDR);
+        owner.grantRole(TIMELOCK_7DAY_DELAY_ROLE, L1_7D_TIMELOCK_ADDR);
+
+        owner.grantRole(bytes32(0), L1_14D_TIMELOCK_ADDR);
+        owner.revokeRole(bytes32(0), vm.addr(L1_DEPLOYER_PRIVATE_KEY));
+    }
+
+    function configProxyAdmin() internal {
+        bytes4[] memory _selectors;
+
+        // no delay, security council
+        _selectors = new bytes4[](3);
+        _selectors[0] = ProxyAdmin.upgradeAndCall.selector;
```

Comment:
[`ProxyAdmin`](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/v4.9.2/contracts/proxy/transparent/ProxyAdmin.sol) has the following functions controlled by `onlyOwner`: 
- `changeProxyAdmin`
- `upgrade`
- `upgradeAndCall`

We should put it all into the selectors.